README.md

```diff
@@ -72,3 +73,70 @@ A data source backend plugin consists of both frontend and backend components.
    ```bash
    mage -l
    ```
+## Authentication
+
+### Local or self-hosted
+
+For a local or self-hosted EdgeDB instance your grafana datasource
+file will look like:
+
+```
+---
+apiVersion: 1
+
+datasources:
+  - name: Metrics (edgedb.com)
+    type: edgedb
+    uid: edgedb_site_metrics
+    access: proxy
+    isDefault: true
+    editable: false
+    jsonData:
+      host: <edgedb-hostname>
+      port: <edgedb-port-number>
+      user: <edgedb-metrics-user>
+      database: <edgedb-metrics-database>
+      tlsSecurity: <"strict"|"insecure"|"no_host_verification">
+      tlsCA: <optional root cert if self-signed>
+    secureJsonData:
+      password: <edgedb-metrics-users-password>
+```
+
+### EdgeDB Cloud
+
+The [edgedb-go](https://github.com/edgedb/edgedb-go) client library
+authenticates [EdgeDB SAAS hosted Cloud](https://www.edgedb.com/cloud)
+differntly than local or self-hosted.  The EdgeDB Cloud instances
+authenticate with a combination of an instance name and a cloud secret
+key.  These are obtained from the EdgeDB Cloud UI when you [create an
+instance](https://cloud.edgedb.com/org/edgedb/create-instance).
+
+The grafana datasource must indicate that the cloud authentication
+settings are in use.  A sample [grafana datasource
+file](https://grafana.com/docs/grafana/latest/datasources) is shown
+below.
+
+The EdgeDB connection uses TLS.  The server presents a certificate
+signed by a widely trusted root so the `tlsCA` parameter should not be
+needed.
+
+```
+---
+apiVersion: 1
+
+datasources:
+  - name: <My Instance> (edgedb cloud)
+    type: edgedb
+    uid: edgedb_site_metrics
+    access: proxy
+    isDefault: true
+    editable: false
+    jsonData:
+      cloudInstance: <my-edge-db-instance-name>
+      database: <my-edgedb-schema-name>
+      tlsSecurity: strict
+    secureJsonData:
+      cloudSecret: <my-instance-cloud-secret>
```

Comment:
Sorry for nit picking! Can this be `secretKey` everywhere? Connection parameter names are standardized across clients etc. so far.